extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -650,7 +650,9 @@ export class ApplicationInsights extends BaseTelemetryPlugin implements IAppInsi
                         _window.addEventListener(_self.config.namePrefix + "locationchange", () => {
                             if (_properties && _properties.context && _properties.context.telemetryTrace) {
                                 _properties.context.telemetryTrace.traceID = generateW3CId();
-                                _properties.context.telemetryTrace.name = _location && _location.pathname || "_unknown_";
+                                (_location && _location.pathname && _location.hash) ? 
```

Comment:
Hmm, way to complicated and hard to read -- also not very minifiable. What about doing this instead
```
let traceLocationName = "_unknown_";
if (_location && _location.pathname) {
    traceLocationName = _location.pathname + (_location.hash || "");
}

_properties.context.telemetryTrace.name = traceLocationName;
```